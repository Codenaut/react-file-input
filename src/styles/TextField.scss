@import "global";

.brainhub-text-field {
  margin: 4px 0;
}

.brainhub-text-field__label {
  font-size: 16px;
  line-height: 18px;
  color: $BROWN_GRAY;
  margin-bottom: 4px;
}

.brainhub-text-field__input {
  font-size: 16px;
  border-radius: 5px;
  padding: 8px 10px;
  border: 1px solid $NEUTRAL_GRAY;
  width: calc(100% - 22px);
}

.brainhub-text-field__input::placeholder {
  color: $NEUTRAL_GRAY;
}
