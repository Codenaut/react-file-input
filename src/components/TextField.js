import React from 'react';
import PropTypes from 'prop-types';

import TextInput from './TextInput';

import '../styles/TextField.scss';

const TextField = ({ label, placeholder, value, onChange }) => (
  <div className="brainhub-text-field">
    <div className="brainhub-text-field__label">{label}</div>
    <div>
      <TextInput placeholder={placeholder} value={value} onChange={onChange}/>
    </div>
  </div>
);

TextField.defaultProps = {
  value: '',
};

TextField.propTypes = {
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

export default TextField;
