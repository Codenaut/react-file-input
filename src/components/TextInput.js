import React from 'react';
import PropTypes from 'prop-types';

import '../styles/TextField.scss';

const TextInput = ({ placeholder, value, onChange }) => (
  <input
    className="brainhub-text-field__input"
    placeholder={placeholder}
    type="text"
    value={value}
    onChange={onChange}
  />
);

TextInput.defaultProps = {
  value: '',
};

TextInput.propTypes = {
  placeholder: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

export default TextInput;
