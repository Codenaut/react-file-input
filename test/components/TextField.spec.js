import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TextField from '../../src/components/TextField';
import TextInput from '../../src/components/TextInput';

const defaultProps = {
  label: 'Test Label',
  placeholder: 'Test Placeholder',
  onChange: () => null,
};

const setup = (props = {}) => {
  const textField = shallow(<TextField {...defaultProps} {...props} />);

  return {
    textField,
    label: textField.find('.brainhub-text-field__label'),
    input: textField.find(TextInput),
  };
};

describe('components', () => {
  describe('TextField', () => {
    it('should render a label', () => {
      const { label } = setup();

      expect(label).toHaveLength(1);
      expect(label.text()).toBe(defaultProps.label);
    });

    it('should render a TextInput', () => {
      const { input } = setup();

      expect(input).toHaveLength(1);
      expect(input.prop('placeholder')).toBe(defaultProps.placeholder);
      expect(input.prop('onChange')).toBe(defaultProps.onChange);
    });

    it('should match exact snapshot', () => {
      const tree = renderer.create(
        <div>
          <TextField {...defaultProps} />
        </div>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
