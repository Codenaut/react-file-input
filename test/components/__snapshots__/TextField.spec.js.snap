// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TextField should match exact snapshot 1`] = `
<div>
  <div
    className="brainhub-text-field"
  >
    <div
      className="brainhub-text-field__label"
    >
      Test Label
    </div>
    <div>
      <input
        className="brainhub-text-field__input"
        onChange={[Function]}
        placeholder="Test Placeholder"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
